controller/controller.go

```diff
@@ -348,7 +352,9 @@ func (c *notificationController) processQueueItem() (processNext bool) {
 		logEntry.Errorf("Failed to process: %v", err)
 		return
 	}
-	if !reflect.DeepEqual(app.GetAnnotations(), appCopy.GetAnnotations()) {
+
+	if !(app.GetAnnotations() == nil && len(appCopy.GetAnnotations()) == 0) &&
```

Comment:
Look like this catches only one case: before annotations map is nil and empty after. Due to refactoring we might end up with a situation where first map is empty and second is nil. Please create a method that compares two maps and reliably compare maps with both cases

Reply:
Good suggestion. Fix is checked in.